contrib_bots/lib/gcalendar.py

```
@@ -0,0 +1,233 @@
+# (through a different option) with the desired calendar.
+# 
+# Something like:
+#   $ ./run.py lib/gcalendar.py -l
+#       <list of calendars and ID's>
+#   $ ./run.py lib/gcalendar.py -c my_cal@group.calendar.google.com
+#       <the bot runs>
+def list_calendars():
+    '''Lists all the calendars with their corresponding ID's.
+
+    This is useful because calendar ID's are a bit of a hassle to
+    get, and are long, random strings that look like an email address.
+
+    The primary calendar is marked as "(primary)"
+    '''
+    credentials = get_credentials()
+    http = credentials.authorize(httplib2.Http())
+    service = discovery.build('calendar', 'v3', http=http)
+
+    cal_response =  service.calendarList().list().execute();
+    calendar_list = cal_response.get('items', []);
+
+    for calendar in calendar_list:
+        name = calendar['summary']
+        cid = calendar['id']
+
+        if 'primary' in calendar:
+            name += ' (primary)'
+
+        print(name + ' - ID: ' + cid)
+
+def reply_message(msg, original_msg, client):
+    '''Replies to a specific message, through the channel of origin.
+
+    Keyword arguments:
+    msg -- content of the reply
+    original_msg -- message that is being replied
+    client -- client from the message handler
+    '''
+    message_type = original_msg['type']
+
+    if message_type == 'private':
+        original_sender = original_msg['sender_email']
+
+        client.send_message(dict(
+            type=message_type,
+            to=original_sender,
+            content=msg,
+        ))
+    elif message_type == 'stream':
+        original_stream = original_msg['display_recipient']
+        original_subject = original_msg['subject']
+
+        client.send_message(dict(
+            type=message_type,
+            to=original_stream,
+            subject=original_subject,
+            content=msg,
+        ))
+    else:
+        raise 'Unknown message type'
```

This should be `raise Exception('Unknown message type')`.